core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -255,4 +257,18 @@ private function overwriteImageTargetDir(array $config, ContainerBuilder $contai
 
         @trigger_error('Using the "contao.image.target_path" parameter has been deprecated and will no longer work in Contao 5.0. Use the "contao.image.target_dir" parameter instead.', E_USER_DEPRECATED);
     }
+
+    private function handleTokenCheckerConfig(array $config, ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition('contao.security.token_checker')) {
+            return;
+        }
+
+        $tokenChecker = $container->getDefinition('contao.security.token_checker');
+        $tokenChecker->replaceArgument(5, new Reference('security.access.simple_role_voter'));
+
+        if ($container->hasParameter('security.role_hierarchy.roles') && \count($container->getParameter('security.role_hierarchy.roles')) > 0) {
```

Comment:
What if there is a `security.role_hierarchy.roles` parameter and it is left empty on purpose? Is it really correct to use `contao.security.token_checker` in this case? Or should we always use `security.role_hierarchy.roles` if it has been set?

Reply:
If an empty hierarchy is defined, the `security.access.role_hierarchy_voter` is removed:

https://github.com/symfony/symfony/blob/8397eb79e003c52b93339c34d9bcafe4848e82c2/src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php#L167-L171